content/en/blog/_posts/2022-04-28-cni-version-related-breaking-change.md

```diff
@@ -0,0 +1,142 @@
+---
+layout: blog
+title: "Breaking changes related to CNI version in popular container runtimes"
+date: 2022-04-28T12:00:00+0000
+slug: cni-version-related-breaking-change-k8s-1.24
+---
+
+**Authors:** Mike Brown (IBM), Rolfe Dlugy-Hegwer (Red Hat), and others.
+
+---
+
+_TL;DR: Before you upgrade to Kubernetes 1.24, if you use a CNI network plugin
+that is set up to use Container Network Interface (CNI) versions earlier than
+v1.0.0 and you plan to use any of the following container runtimes, you must
+specify the CNI version. Otherwise, your network will break._
+
+This issue applies to the following container runtimes:
+- **containerd** v1.6.0 and later
+- **CRI-O** 1.24 and later
+
+Here’s an overview of the typical steps to perform on each node:
+
+1. [Safely drain and cordon the node](/docs/tasks/administer-cluster/safely-drain-node/).
+2. After stopping your container runtime and kubelet services, perform the
+following upgrade operations:
+  - If you’re running CNI plugins, upgrade them to the latest version.
```

Comment:
Why is this step needed? We don't make that clear.

Reply:
well that would depend on which plugins you are using, whether they are in support, which version of the cni libraries they were built against, what config files you have for which pods you want to keep running, which bug fixes one may need, and whether to stay on pre 1.0 releases ..  but yes it's not an order .. just a recommendation